src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> current : table) {
+            while (current != null) {
+                if (Objects.equals(key, current.key)) {
+                    return current.value;
+                }
+                current = current.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        int index = getTableIndex(table.length, hash);
+        Node<K, V> current = table[index];
+        Node<K, V> newNode = new Node<>(hash, key, value);
+        if (current == null) {
+            table[index] = newNode;
+        } else {
+            while (true) {
+                if (Objects.equals(key, current.key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    break;
```

Comment:
In my opinion, you need to change the code a bit here so that you don't write the "while (true)" condition, but write the condition for interrupting the loop in brackets. But let's ask the mentor for advice, shall we?